src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/ReferenceDataFixesService.java

```diff
@@ -0,0 +1,123 @@
+import uk.gov.hmcts.ecm.common.exceptions.CaseRetrievalException;
+import uk.gov.hmcts.ecm.common.model.ccd.CCDRequest;
+import uk.gov.hmcts.ecm.common.model.ccd.CaseData;
+import uk.gov.hmcts.ecm.common.model.ccd.SubmitEvent;
+import uk.gov.hmcts.ecm.common.model.ccd.items.HearingTypeItem;
+import uk.gov.hmcts.ecm.common.model.ccd.types.HearingType;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.*;
+import uk.gov.hmcts.ethos.replacement.docmosis.service.refdatafixes.refData.RefDataFixesData;
+import uk.gov.hmcts.ethos.replacement.docmosis.service.refdatafixes.refData.RefDataFixesDetails;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service("referenceDataFixesService")
+public class ReferenceDataFixesService {
+    private static final String CASES_SEARCHED = "Cases searched: ";
+    private static final String MESSAGE = "Failed to retrieve reference data for case id : ";
+    private final CcdClient ccdClient;
+
+    public RefDataFixesData updateJudgesItcoReferences(RefDataFixesDetails refDataFixesDetails, String authToken, RefDataFixesCcdDataSource dataSource) {
+
+        RefDataFixesData refDataFixesData = refDataFixesDetails.getCaseData();
+        String existingJudgeCode = refDataFixesData.getExistingJudgeCode();
+        String requiredJudgeCode = refDataFixesData.getRequiredJudgeCode();
+        String caseTypeId = refDataFixesDetails.getCaseTypeId().contains("_") ?
+                Strings.split(refDataFixesDetails.getCaseTypeId(), "_")[0] : refDataFixesDetails.getCaseTypeId();
+        List<String> dates = getDateRangeForSearch(refDataFixesDetails);
+        String dateFrom = dates.get(0);
+        String dateTo = dates.get(1);
+        try {
+            List<SubmitEvent> submitEvents = dataSource.getData(caseTypeId, dateFrom, dateTo, ccdClient);
+            if (CollectionUtils.isNotEmpty(submitEvents)) {
+                log.info(CASES_SEARCHED + submitEvents.size());
+                for (SubmitEvent submitEvent : submitEvents) {
+                    CaseData caseData = submitEvent.getCaseData();
+                    setJudgeName(caseData, existingJudgeCode, requiredJudgeCode);
+
+                    CCDRequest returnedRequest = ccdClient.startEventForCase(authToken, caseTypeId,
+                            refDataFixesDetails.getJurisdiction(), String.valueOf(submitEvent.getCaseId()));
+                    ccdClient.submitEventForCase(authToken, caseData, caseTypeId,
+                            refDataFixesDetails.getJurisdiction(), returnedRequest, String.valueOf(submitEvent.getCaseId()));
+                }
+                log.info(String.format(
+                        "Existing Judge's code in all cases from %s to %s is " +
+                                "updated to required judge's code", dateFrom, dateTo));
+            }
+            return refDataFixesData;
+        } catch (Exception ex) {
+            log.error(MESSAGE + refDataFixesDetails.getCaseId(), ex);
```

Comment:
Logging and throwing is an anti-pattern. What do you see in the logs by doing this? Do you get the same thing twice?

Reply:
removed throwing error